peer/peer.go

```diff
@@ -1493,6 +1528,28 @@ out:
 			}
 			break out
 		}
+
+		// Send the received message to all the subscribers.
+		for sub := range p.subscribers {
+			select {
+			case <-sub.quit:
+				delete(p.subscribers, sub)
+				continue
+			default:
+			}
+
+			// Spin up a goroutine so that we don't block here.
+			go func(subscription recvMsgsubscription,
```

Comment:
Just wanna note that this will create many short-lived goroutines given it spawns a new goroutine for every single message delivered to every single subscriber - we may consider buffering the subscriber's channel, tho non-blocking.